articles/articles/article.php

```diff
@@ -1,176 +1,331 @@
+			unset($subArray->modified_by_name);
+			unset($subArray->urls);
+			unset($subArray->attribs);
+			unset($subArray->metadata);
+			unset($subArray->metakey);
+			unset($subArray->metadesc);
+			unset($subArray->xreference);
+			unset($subArray->readmore);
+			unset($subArray->category_route);
+			unset($subArray->category_access);
+			unset($subArray->category_alias);
+			unset($subArray->author_email);
+			unset($subArray->parent_title);
+			unset($subArray->parent_id);
+			unset($subArray->parent_route);
+			unset($subArray->parent_alias);
+			unset($subArray->rating);
+			unset($subArray->rating_count);
+			unset($subArray->published);
+			unset($subArray->parents_published);
+			unset($subArray->alternative_readmore);
+			unset($subArray->layout);
+			unset($subArray->params);
+			unset($subArray->displayDate);
+			unset($subArray->author);
+			unset($subArray->category_title);
+		}
+
+		$obj = new stdclass;
+
+		if (count($rows) > 0)
+		{
+			$obj->success = 'true';
+			$obj->{'data'} = array('results' => $rows, 'total' => count($rows));
+
+			return $obj;
+		}
+		else
+		{
+			$obj->success = 'false';
+			$obj->message = 'System does not have articles';
+		}
+
+		return $obj;
 	}
-	
+
+	/**
+	 * Post is to create / upadte article
+	 *
+	 * @return  Bolean
+	 *
+	 * @since  3.5
+	 */
 	public function post()
-	{  
-		//init variable
+	{
+		$this->plugin->setResponse($this->CreateUpdateArticle());
+	}
+
+	/**
+	 * CreateUpdateArticle is to create / upadte article
+	 *
+	 * @return  Bolean
+	 *
+	 * @since  3.5
+	 */
+	public function CreateUpdateArticle()
+	{
+		if (version_compare(JVERSION, '3.0', 'lt'))
+		{
+			JTable::addIncludePath(JPATH_PLATFORM . 'joomla/database/table');
+		}
+
+		$obj = new stdclass;
+
 		$app = JFactory::getApplication();
-		
-		$result = new stdClass();
-		$article_id	= $app->input->get('id', 0, 'INT');
-		
-		if(!$article_id)
-		{
-			$result->success = 0;
-			$result->message = 'Please select article';
-			$this->plugin->setResponse($result);
-			return;
-		}
-		
-		$art_obj = new ContentModelArticle();
-		$art_obj->setId($article_id);
-		$a_data = $art_obj->getArticle();
-		
-		//format data
-		$obj = new BlogappSimpleSchema();
-		//$item = $obj->mapPost($a_data,'', 100, array('text'));
-		$item = $obj->mapPost($a_data,'', 100, array());
-	
-		$this->plugin->setResponse($item);
+		$article_id = $app->input->get('id', 0, 'INT');
+
+		if (empty($app->input->get('title', '', 'STRING')))
+		{
+			$obj->success = 'false';
+			$obj->message = 'Title is Missing';
+
+			return $obj;
+		}
+
+		if (empty($app->input->get('introtext', '', 'STRING')))
+		{
+			$obj->success = 'false';
+			$obj->message = 'Introtext is Missing';
+
+			return $obj;
+		}
+
+		if (empty($app->input->get('catid', '', 'INT')))
+		{
+			$obj->success = 'false';
+			$obj->message = 'Category id is Missing';
+
+			return $obj;
+		}
+
+		if ($article_id)
+		{
+			$article = JTable::getInstance('Content', 'JTable', array());
```

Comment:
@vijaykhollam Have you verified that when you create an article via the API you are able to see it correctly in the admin and also in the frontend ?

Yes. I have verified this.